### PR TITLE
Provide the distributed smoke test with rabit of xgboost

### PR DESCRIPTION
**Pull request by merlintang**
## what is proposed in this PR 
Provide the distributed rabit test example 

## how to test it 
Test with minikube locally.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost-operator/25)

### CODE REVIEW

File: config/samples/smoke-dist/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+numpy>=1.16.3
+Cython>=0.29.4
+requests>=2.21.0
+urllib3>=1.21.1
+pandas>=0.23.2
```
Comment (terrytangyuan) [inline config/samples/smoke-dist/requirements.txt]: pandas (and maybe others) doesn’t seem used.

Comment (merlintang) [reply config/samples/smoke-dist/requirements.txt]: I would pandas later for transform the input data into dmatrix. thanks terry.

Comment (terrytangyuan) [reply config/samples/smoke-dist/requirements.txt]: Then I'd suggest adding them later when needed.

File: config/samples/smoke-dist/tracker.py [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+Tracker script for DMLC
+Implements the tracker control protocol
+ - start dmlc jobs
+ - start ps scheduler and rabit tracker
+ - help nodes to establish links with each other
+Tianqi Chen
+Notes from Matthew Rocklin
+--------------------------
+This was taken from
+https://github.com/dmlc/dmlc-core/blob/master/tracker/dmlc_tracker/tracker.py
```
Comment (terrytangyuan) [inline config/samples/smoke-dist/tracker.py]: You can remove the notes from Matthew and just keep this link to dmlc-core.

File: pkg/controller/xgboostjob/util.go [hunk 3]
```diff
@@ -87,6 +92,47 @@ func computeTotalReplicas(obj metav1.Object) int32 {
...
+}
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	return os.Getenv("USERPROFILE") // windows
+}
+
+func isGangSchedulerSet(replicas map[common.ReplicaType]*common.ReplicaSpec) bool {
```
Comment (terrytangyuan) [inline pkg/controller/xgboostjob/util.go]: You can reuse this from kubeflow/common

**Comment by johnugeorge** [comment]
Please consider adding  presubmit tests

**Comment by merlintang** [comment]
@johnugeorge thanks for your comments, I would propose a follow up pr for this.

**Pull request merged**